jinahub/rankers/DPRReaderRanker/dpr_reader.py

```diff
@@ -0,0 +1,231 @@
+from itertools import groupby
+from typing import Dict, Iterable, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+from jina import Document, DocumentArray, Executor, requests
+from transformers import DPRReader, DPRReaderTokenizerFast
+
+
+def _batcher(iterable, n=1):
+    """Batch an iterable. Here temporarily until
+    https://github.com/jina-ai/jina/issues/3068
+    is solved.
+    """
+    l = len(iterable)
+    for ndx in range(0, l, n):
+        yield iterable[ndx : ndx + n]
+
+
+def _logistic_fn(x: np.ndarray) -> List[float]:
+    """Compute the logistic function"""
+    return (np.exp(x) / (1 + np.exp(x))).tolist()
```

Comment:
The old implementation has extra computation and does not work for 
```python
_logistic_fn(np.array([np.inf, 1, 2]))
```

```suggestion
def _logistic_fn(x: np.ndarray) -> List[float]:
    """Compute the logistic function"""
    return (1 / (1 + np.exp(-x))).tolist()
```